io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -556,9 +519,8 @@ public boolean hasNext() {
 		 *         number of results reported by the WFS.
 		 */
 		protected boolean isFeatureLimitReached() {
-			return (maxNumberOfFeatures != UNLIMITED
-					&& totalFeaturesProcessed >= maxNumberOfFeatures)
-					|| (size != UNKNOWN_SIZE && totalFeaturesProcessed >= size);
+			return maxNumberOfFeatures != UNLIMITED
+					&& totalFeaturesProcessed >= maxNumberOfFeatures;
```

Comment:
Won't we still get problems here because the additional objects are counted as features processed?

Reply:
If we just setup the `hits == UNKNOWN_SIZE`, that's the code left. It didn't make sense to me to keep it as long as `size `would also be always `UNKNOWN_SIZE`. 
I have used just 2 projects to test, probably it's not enough, but I haven't encountered any problem.

Reply:
> If we just setup the hits == UNKNOWN_SIZE

> size would also be always UNKNOWN_SIZE

Can you explain why `size` should always be `UNKNOWN_SIZE`? In case the request does not contain a `RESOLVEDEPTH` parameter, the old logic can still be used IMO.

Reply:
From this code
```
                switch (wfsVersion) {
		case "1.1.0":
			// The "numberOfFeatures" reported by a 1.1.0 WFS may be smaller
			// than the actual number of features matches by the query if the
			// number of features returned per query is limited on the server
			// side. Therefore do not rely on it as a size information here.
			this.size = UNKNOWN_SIZE;
			break;
		case "2.0.0":
		case "2.0.2":
			// The "numberMatched" reported by a 2.0.0/2.0.2 WFS should be
			// number of features matched by the query. If hits equals
			// UNKNOWN_SIZE then size is also set to that value
			this.size = isLimited() ? Math.min(maxNumberOfFeatures, hits) : hits;
			break;
		default:
			this.size = UNKNOWN_SIZE;
		}
```
having hits = UNKNOWN_SIZE, also for the case 2.0.2 the size would come out as UNKNOWN_SIZE, independently if isLimited() is true or false.